languages/php/menhir/parsing_hacks_php.ml

```diff
@@ -192,11 +213,21 @@ let find_typehint toks =
 (* Fix tokens *)
 (*****************************************************************************)
 
-let fix_tokens xs =
+let fix_tokens xs ~keep_func_doc =
   let rec aux env acc xs =
     match xs with
     (* need an acc, to be tail recursive, otherwise get some stack overflow *)
     | [] -> List.rev acc
+    (* If the following token (when ignoring whitespace/comments) is a func_dec,
+       the doc_comment should be treated differently optionally depending on
```

Comment:
Nit: "... treated differently optionally depending ..." is a bit dense, and the comment does not clarify very well what's going on. If `keep_func_doc` is enabled, and we find a doc-comment that precedes a function declaration, we'll rewrite the token as `T_FUNC_DOC_COMMENT` and the parser will keep it instead of discarding it.